sunset/code/modules/salvage/structures.dm

```diff
@@ -71,16 +119,67 @@
 	name = "[material] Material Processor"
 	icon_state = "material_processor_[material]"
 
+//===================
+//makeshift generator
+//===================
+
 /obj/machinery/makeshift/generator //like a pacman except it runs off burnables
 	material = "Wooden"
 	name = "error"
 	desc = "Makes some power for your machines and wakes the neighbors. You should really put a proper muffler on it."
 	icon_state = "error"
+	amchored = FALSE
+	use_power = NO_POWER_USE
+
+	var/active = 0
+	var/power_gen = 3500
+	var/recent_fault = 0
+	var/consumption = 0
+
+	interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND | INTERACT_ATOM_UI_INTERACT | INTERACT_ATOM_REQUIRES_ANCHORED
 
 /obj/machinery/makeshift/generator/Initialize()
 	. = ..()
 	name = "[material] Generator"
-	icon_state = "generator_[material]"
+	icon_state = "generator_[material]_[active]"
+	soundloop = new(list(src), active)
+
+/obj/machinery/makeshift/generator/proc/HasFuel()
+	return 1
+
+/obj/machinery/makeshift/generator/proc/UseFuel()
+	return
+
+/obj/machinery/makeshift/generator/proc/DropFuel()
+	return
+
+/obj/machinery/makeshift/generator/proc/handleInactive()
+	return
+
+/obj/machinery/power/port_gen/update_icon()
+	icon_state = "generator_[material]_[active]"
+
+/obj/machinery/power/port_gen/process()
+	if(active && HasFuel() && !crit_fail && anchored && powernet)
+		add_avail(power_gen * power_output)
+		UseFuel()
+		src.updateDialog()
+		soundloop.start()
+
+	else
+		active = 0
+		handleInactive()
+		update_icon()
+		soundloop.stop()
+
+/obj/machinery/power/port_gen/examine(mob/user)
+	..()
+	to_chat(user, "It is[!active?"n't":""] running.")
```

Comment:
This is confusing for nothing.Simplify it
```suggestion
	to_chat(user, "It [active ? "is" : "isn't"] running.")
```

Reply:
generator code, removed for time being